src/NewlineLexers.jl

```diff
@@ -429,22 +503,32 @@ function _find_newlines_quote_unaware_simd!(l::Lexer, buf::Vector{UInt8}, out::A
     end
 end
 
+"""
+    find_newlines!(l::Lexer, buf::Vector{UInt8}, out::AbstractVector{Int32}, curr_pos::Int=firstindex(buf), end_pos::Int=lastindex(buf))
```

Comment:
it would be idiomatic for the argument being mutated `out` to be the first argument. 

Is there a reason to break from that idiom?

Can we open an issue to change to that, so we have 
```julia
find_newlines!(out::AbstractVector{Int32}, l::Lexer, buf::Vector{UInt8}, curr_pos::Int=firstindex(buf), end_pos::Int=lastindex(buf))
```
?

Reply:
I think you are right, but it does feel weird to me that the thing that impacts the implementation the most and that also gets mutated internally comes second... In an OO language, this would be `Lexer.find_newlines(...)`, I think this is what throws me off.